t/backup_expiration.t

```diff
@@ -0,0 +1,250 @@
+use Test::Spec;
+
+use HTFeed::BackupExpiration;
+use HTFeed::Config qw(set_config get_config);
+use HTFeed::DBTools qw(get_dbh);
+use HTFeed::Storage::ObjectStore;
+use HTFeed::Storage::PrefixedVersions;
+
+use strict;
+
+describe "HTFeed::BackupExpiration" => sub {
+  spec_helper 'storage_helper.pl';
+  spec_helper 's3_helper.pl';
+  local our ($tmpdirs, $testlog, $bucket, $s3);
+  
+  my $old_storage_classes;
+
+  before each => sub {
+    $old_storage_classes = get_config('storage_classes');
+    my $new_storage_classes = {
+      'prefixedversions-test' =>
+      {
+        class => 'HTFeed::Storage::PrefixedVersions',
+        obj_dir => $tmpdirs->{backup_obj_dir},
+        encryption_key => $tmpdirs->test_home . "/fixtures/encryption_key"
+      },
+      'objectstore-test' =>
+      {
+        class => 'HTFeed::Storage::ObjectStore',
+        bucket => $s3->{bucket},
+        awscli => $s3->{awscli},
+        encryption_key => $tmpdirs->test_home . "/fixtures/encryption_key"
+      }
+    };
+    set_config($new_storage_classes,'storage_classes');
+  };
+
+  after each => sub {
+    set_config($old_storage_classes,'storage_classes');
+  };
+
+  sub staged_volume_storage {
+    my $storage_name = shift;
+
+    my $volume = stage_volume($tmpdirs, @_);
+    my $storage_config = get_config('storage_classes')->{$storage_name};
+    my $storage = $storage_config->{class}->new(volume => $volume,
+                                                config => $storage_config,
+                                                name   => $storage_name);
+    return $storage;
+  }
+
+  sub old_random_timestamp {
+    my $sql = 'SELECT DATE_FORMAT(DATE_SUB(NOW(), INTERVAL (180 + (RAND() * 180)) DAY),"%Y%m%d%H%i%S")';
+    my @res = get_dbh->selectrow_array($sql);
+    return $res[0];
+  }
+
+  sub new_random_timestamp {
+    my $sql = 'SELECT DATE_FORMAT(DATE_SUB(NOW(), INTERVAL (RAND() * 180) DAY),"%Y%m%d%H%i%S")';
+    my @res = get_dbh->selectrow_array($sql);
+    return $res[0];
+  }
+
+  describe "HTFeed::BackupExpiration for PrefixedVersions" => sub {
+    describe "#new" => sub {
+      it "succeeds" => sub {
+        my $exp = HTFeed::BackupExpiration->new(storage_name => 'prefixedversions-test');
+        ok($exp, 'new returns a value');
+      };
+    };
+
+    describe "#run" => sub {
+      it "does not do anything with a single version" => sub {
+        my $storage = staged_volume_storage('prefixedversions-test', 'test','test');
+        my $version = old_random_timestamp();
```

Comment:
This sequence shows up several times in these tests (as well as for objectstore where `stage`/`make_object_path` is a no-op); I would consider extracting these to a helper method and perhaps seeing if there's any other repeated setup in these tests that might be clearer or easier to maintain if we extracted it